Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -499,12 +499,75 @@ def _run_single_test(self, driver, test_input, stop_when_done):
+        'imported/w3c/web-platform-tests/content-security-policy': 1095,
+        'imported/w3c/web-platform-tests/css/css-grid/placement': 1035,
+        'imported/w3c/web-platform-tests/css/css-transitions': 821,
+        'imported/w3c/web-platform-tests/css/geometry': 1476,
+        'imported/w3c/web-platform-tests/dom/events': 1568,
+        'imported/w3c/web-platform-tests/feature-policy': 2609,
+        'imported/w3c/web-platform-tests/fetch': 423,
+        'imported/w3c/web-platform-tests/html/anonymous-iframe': 1373,
+        'imported/w3c/web-platform-tests/html/browsers/the-window-object': 732,
+        'imported/w3c/web-platform-tests/html/canvas/element/manual': 675,
+        'imported/w3c/web-platform-tests/html/cross-origin-embedder-policy': 1653,
+        'imported/w3c/web-platform-tests/html/cross-origin-opener-policy': 10639,
+        'imported/w3c/web-platform-tests/html/semantics/document-metadata': 740,
+        'imported/w3c/web-platform-tests/html/semantics/embedded-content': 978,
+        'imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0': 519,
+        'imported/w3c/web-platform-tests/html/semantics/forms/textfieldselection': 1021,
+        'imported/w3c/web-platform-tests/html/semantics/scripting-1': 1303,
+        'imported/w3c/web-platform-tests/html/semantics/scripting-1/the-script-element/css-module': 5601,
+        'imported/w3c/web-platform-tests/html/semantics/scripting-1/the-script-element/moving-between-documents': 5946,
+        'imported/w3c/web-platform-tests/html/syntax/speculative-parsing': 2799,
+        'imported/w3c/web-platform-tests/media-source': 6747,
+        'imported/w3c/web-platform-tests/mediacapture-record': 12018,
+        'imported/w3c/web-platform-tests/mediacapture-streams': 856,
+        'imported/w3c/web-platform-tests/pointerevents': 3544,
+        'imported/w3c/web-platform-tests/preload': 1316,
+        'imported/w3c/web-platform-tests/reporting': 738,
+        'imported/w3c/web-platform-tests/requestidlecallback': 789,
+        'imported/w3c/web-platform-tests/uievents': 2401,
+        'imported/w3c/web-platform-tests/webrtc': 24445,
+        'imported/w3c/web-platform-tests/webrtc-encoded-transform': 11348,
+        'imported/w3c/web-platform-tests/xhr': 1002,
+        'jquery': 842,
+        'media': 436,
+        'platform': 804,
+        'model-element': 1390,
+        'swipe': 10968,
+        'webgl': 533,
+        'webrtc': 14712,
+        'workers': 1389,
+    }
 
     def __init__(self, name, test_inputs):
         self.name = name
         self.test_inputs = test_inputs
         self.needs_servers = test_inputs[0].needs_servers
 
+        parts = name.split('/')
+        key = parts[0]
+        weight = self.WEIGHT_ESTIMATES.get(key, self.DEFAULT_WEIGHT)
+        for part in parts[1:]:
+            key = '{}/{}'.format(key, part)
+            weight = self.WEIGHT_ESTIMATES.get(key, weight)
+
+        self.cost = len(self.test_inputs) * weight
```

Comment:
Does this not end up with the weight of those which we explicitly have estimates for being n*2 for n tests?

Reply:
Weight estimates aren't number of tests, they're average number of milliseconds per test. The "cost" of a shard is basically estimated runtime (which is probably a better name than cost), and is computed by our guess for the average runtime of a test in that shard multiplied by the number of tests in the shard.